rs/messaging/src/routing/stream_handler.rs

```diff
@@ -610,14 +610,14 @@ impl StreamHandlerImpl {
                         state,
                         available_guaranteed_response_memory,
                     ) {
-                        Ok(()) => {
+                        None => {
                             // Reject response successfully inducted or dropped.
                         }
-                        Err((StateError::CanisterMigrating { .. }, reject_response)) => {
+                        Some((StateError::CanisterMigrating { .. }, reject_response)) => {
                             // Canister is being migrated, reroute reject response.
                             reroute_response(reject_response, state, streams, &self.log);
                         }
```

Comment:
Is it really possible that a response that is already rerouted needs to be rerouted again?

Reply:
Strike that - I read that as being part of the inline function `reroute_response` above and thought we'd end up with a recursive call.